scipy/spatial/_polygon_surface_area.py

```diff
@@ -0,0 +1,79 @@
+"""
+
+Polygon Surface Area Code
+
+.. versionadded:: 1.0.0
+
+"""
+
+import numpy as np
+from scipy.spatial.distance import pdist
+from . import _surface_area
+
+#
+# Copyright (C)  James Nichols and Tyler Reddy
+# Contributions from Tyler Reddy are owned by LANL
+# and respect the scipy distribution license
+#
+# Distributed under the same BSD license as Scipy.
+
+def _vertex_index_strider(index, num_vertices):
+    # handle the wrapping / iteration over
+    # polygon vertices in either CW or CCW
+    # sort order
+    forward_index = index + 1
+    backward_index = index - 1
+    if forward_index > (num_vertices - 1):
+        forward_index = 0
+    return forward_index, backward_index
+
+def poly_area(vertices, radius=None, threshold=1e-21,
+              cython=None):
```

Comment:
It makes things easier for testing, but for the final API I'm not a fan of the `cython` kwarg here. Once things settle down, we should pick one implementation and use that.